### PR TITLE
feat(ingest): add escape hatch methods to SqlParsingAggregator

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py [hunk 1]
```diff
@@ -284,6 +311,96 @@ def _initialize_schema_resolver_from_graph(self, graph: DataHubGraph) -> None:
...
+
+        Args:
+            known_query_lineage: The known query lineage information.
+        """
+
+        # Generate a fingerprint for the query.
+        query_fingerprint = get_query_fingerprint(
+            known_query_lineage.query_text, self.platform.platform_name
+        )
+        # TODO format the query text?
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py:334]: I think formatting the query might be useful if we don't do it on the frontend side when we are showing the queries, but it should be optional.

File: metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py [hunk 2]
```diff
@@ -284,6 +311,96 @@ def _initialize_schema_resolver_from_graph(self, graph: DataHubGraph) -> None:
...
+
+        # Register the query.
+        self._add_to_query_map(
+            QueryMetadata(
+                query_id=query_fingerprint,
+                formatted_query_string=known_query_lineage.query_text,
+                session_id=_MISSING_SESSION_ID,
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py:341]: Is there a value if we can pass in session id to the know lineage as well?

Comment (treff7es) [reply metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py:341]: In some case I think we should be able to capture

**Review by treff7es — APPROVED** [review]

File: metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py:334]: yeah I will do it in a future PR

File: metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py:341]: yup I'll add that in the next RP

**Pull request merged**